src/odemis/driver/test/smartpod_test.py

```diff
@@ -0,0 +1,168 @@
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+import logging
+import os
+import time
+import unittest
+from odemis.driver import smartpod
+from odemis.util import test
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
+
+# Export TEST_NOHW=1 to force using only the simulator and skipping test cases
+# needing real hardware
+TEST_NOHW = (os.environ.get("TEST_NOHW", 0) != 0)  # Default to Hw testing
+
+COMP_ARGS = {
+    "atol": 1e-3,
+    "rtol": 1e-3,
+    }
+
+CONFIG = {"name": "SmartPod",
+        "role": "",
+        "locator": "usb:ix:0",
+        # "locator": "fake",
+        "options":"",
+        "axes": {
+            'x': {
+                'range': [-1, 1],
+                'unit': 'm',
+            },
+            'y': {
+                'range': [-1, 1],
+                'unit': 'm',
+            },
+            'z': {
+                'range': [-1, 1],
+                'unit': 'm',
+            },
+            'theta_x': {
+                'range': [0, 3.1415],
+                'unit': 'rad',
+            },
+            'theta_y': {
+                'range': [0, 3.1415],
+                'unit': 'rad',
+            },
+            'theta_z': {
+                'range': [0, 3.1415],
+                'unit': 'rad',
+            },
+        },
+}
+
+
+class TestSmartPod(unittest.TestCase):
+    """
+    Tests cases for the SmartPod actuator driver
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.dev = smartpod.SmartPod(**CONFIG)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.dev.terminate()  # free up socket.
+        
+    def test_reference_cancel(self):
+        # Test canceling referencing
+        f = self.dev.reference()
+        time.sleep(0.1)
+        f.cancel()
+
+        for a, i in self.dev.referenced.value.items():
+            self.assertFalse(i)
+
+        f = self.dev.reference()
+        f.result()
+
+        for a, i in self.dev.referenced.value.items():
+            self.assertTrue(i)
+
+    def test_out_of_range(self):
+        """
+        Test sending a position that is out of range.
+        """
+        pos = {'x': 1.5, 'y': 20, 'z': 0, 'theta_x': 0, 'theta_y': 0, 'theta_z': 0.0005}
+        with self.assertRaises(ValueError):
+            self.dev.moveAbs(pos).result()
+
+    def test_move_abs(self):
+        self.dev.SetSpeed(0.1)
```

Comment:
Why do you set the speed? Is that necessary before being able to move? If so, it should probably in the init. Either hard-code the speed for now, or get it as argument.